torch/jit/_script.py

```diff
@@ -879,8 +877,18 @@ def _script_pdt(obj, optimize=None, _frames_up=0, _rcb=None, example_inputs: Opt
     if monkeytype_trace:
         monkeytype_config = JitTypeTraceConfig(type_trace_db)
         with monkeytype_trace(monkeytype_config):
-            for example_input in example_inputs:  # type: ignore[union-attr]
-                obj(*example_input)
+            for example_input in example_inputs:    # type: ignore[union-attr]
+                # If the obj is an nn.Module or a class, then each method is
+                # executed with the arguments provided in the example inputs.
+                # example inputs here will be of tuple(class.method, (arguments))
+                # This is used to infer type annotations for those methods
+                # which are not called directly under the hood of monkeytype.
+                if isinstance(obj, torch.nn.Module) or inspect.isclass(obj):
+                    for module, example_input in example_inputs:  # type: ignore[union-attr]
```

Comment:
Also, it seems that now you are assuming `example_inputs` is of type `List[Tuple[nn.Module, Tuple]]`, this doesn't match annotated type for it. Maybe that's why mypy complains about `union-attr` lint error. Please fix.